test/framework/src/main/resources/log4j2.component.properties

```diff
@@ -0,0 +1,7 @@
+# RandomizedRunner renames threads and we'd like to look at accurate thread
+# names. Also, this is the default in log4j2's master branch for versions of
+# Java after 1.8_102 which most of the versions we use are anyway. After that
+# version of Java calling `Thread.getName()` doesn't allocate so UNCACHED
+# ought to be faster than CACHED because it doesn't have to deal with
+# ThreadLocals.
+AsyncLogger.ThreadNameStrategy = UNCACHED
```

Comment:
Note: We don't use the async logger but we still use mutable log events under the hood and they are configured by this property. I'll add that in a comment when I get a chance.